src/components/SidePanels/OutlinerPanel/OutlinerTree.tsx

```diff
@@ -129,13 +129,33 @@ const useDataTree = ({
     const style = item?.background ?? item?.getOneColor?.();
     const color = chroma(style ?? '#666').alpha(1);
     const mods: Record<string, any> = { hidden, type, isDrawing };
+
     const label = (() => {
       if (!type) {
         return 'No Label';
       } else if (type.includes('label')) {
         return item.value;
       } else if (type.includes('region') || type.includes('range')) {
-        return (item?.labels ?? []).join(', ') || 'No label';
+        const labelsInResults = item.results
+          .filter((result: any) => result.type.endsWith('labels'))
+          .map((result: any) => result.selectedLabels || []);
+
+        const labels: any[] = [].concat(...labelsInResults);
```

Comment:
it's simplier to do this via `[... labelsInResults]`, but you don't need this at all — `labelsInResults` is already a copy

Reply:
I have to do this in this case because sometimes labelsInResult is returning 2 arrays or more arrays with arrays inside with more than 1 node, so here I'm breaking the parent array and concatenating the children arrays in just 1 array, if I do [...labelInResult] the result can be [array(1), array(3)]

Reply:
oh, yes, sorry, missed that! so you can use `.flat(1)` for this, a bit more clear than `concat`